### DISCUSSION TITLE
Compare RDA entities with similar entities from other sources

### DISCUSSION THREAD

**Discussion by pennylenger** [crystalyragui/MARC2RDA#464] [Q&A]
I'm working on comparing similar entities from other sources to RDA entities based on Adam's [WikiProject URIs in MARC](https://www.wikidata.org/wiki/Wikidata:WikiProject_URIs_in_MARC) table. I check the class/entity/type of different sources of values for subfields $0 and $1 to determine whether we can safely use that URI. In my table [Entity Type of URIs in MARC](https://docs.google.com/spreadsheets/d/1p_lDvrR58zD7ZN0_QM1vaPvr06rvelA6cB0piCZB6es/edit?gid=0#gid=0) (including all analysis process), the header lists the RDA entity types and their definitions. In each row, I've listed similar entities from other sources, their definition and determined whether they are a supertype, equivalent or subtype of that RDA entity. For some sources, I am able to find clear definitions, while for those I can't find a definition, I look at their hierarchies and properties to make a determination. These are just my interpretations and might not be accurate, so I would appreciate your feedback and corrections. [Approved URI](https://docs.google.com/spreadsheets/d/1ryMjPPLD6QyRx2vEruxqR7xWYNinaZjA1DWDuLr4ltI/edit?gid=0#gid=0) (only including entity type uri)

**Comment by pennylenger** [comment]
Hi @AdamSchiff For LCNAF, it is an authority file, but we can also get rwo URI. I want to find a similar rda:person entity. I first looked at an authority record [Twain, Mark, 1835-1910 (Spirit)](https://id-loc-gov.offcampus.lib.washington.edu/authorities/names/n82045653.html) and found it is an instance of [MADS/RDF PersonalName](http://www.loc.gov.offcampus.lib.washington.edu/mads/rdf/v1#PersonalName), [MADS/RDF Authority](http://www.loc.gov.offcampus.lib.washington.edu/mads/rdf/v1#Authority) and [SKOS Concept](http://www.w3.org/2004/02/skos/core#Concept). Is madsrdf:PersonalName equivalent to rda:person? It seems fictional. 

Then I looked at this rwo record  [Angelou, Maya](https://id-loc-gov.offcampus.lib.washington.edu/rwo/agents/n50024879.html) and found it is Instance Of [MADS/RDF RWO](http://www.loc.gov.offcampus.lib.washington.edu/mads/rdf/v1#RWO), [BF Person](http://id.loc.gov.offcampus.lib.washington.edu/ontologies/bibframe/Person), [FOAF Person](http://xmlns.com/foaf/0.1/Person). FOAF person has a clear definition: The [Person](http://xmlns.com/foaf/spec/#term_Person) class represents people. Something is a [Person](http://xmlns.com/foaf/spec/#term_Person) if it is a person. We don't nitpic about whether they're alive, dead, real, or imaginary. So person in LCNAF is supertype of rda:person? 

Where can I find a definition of person in LCNAF?  Thank you.

**Reply by CECSpecialistI** [reply]
Mark Twain (Spirit) might not be defined as person because it's also a pseudonym?

**Reply by CECSpecialistI** [reply]
I think that's the answer. This spirit is defined as a person. https://id.loc.gov/rwo/agents/nb2010022625.html

**Reply by CECSpecialistI** [reply]
It's just not for pseudonyms. Mark Twain was a pseudonym for [this person](http://id.loc.gov/rwo/agents/n93099439)

**Reply by pennylenger** [reply]
Thank you Crystal. Do you know for wikidata, physical location (Q17334923): position of something in space or geographic location (Q2221906): location of a point or an area on something's surface or elsewhere, which one is equivalent to rda:place: A given extent of space?

**Reply by CECSpecialistI** [reply]
That's really tough. I don't think either of those. I think it's this one: http://www.wikidata.org/entity/Q27096213

**Reply by CECSpecialistI** [reply]
Ooh, yes it is it has a statement that it has equivalent class of LRM:Place. http://iflastandards.info/ns/lrm/lrmer/E10

**Reply by CECSpecialistI** [reply]
I think in Wikidata if we have instances of subclasses of Q27096213 they are probably fine as rda:Places? @GordonDunsire ?

**Reply by GordonDunsire** [reply]
On person:

madsrdf:PersonalName has the definition '... a resource whose label represents a personal name'. I assume this means an entity with a name that is similar to a conventional name for a person. The definition is basically circular: is 'Alice Cooper' a personal name? is 'The artist formerly known as Prince' a personal name? My conclusion is that the semantics of 'Person' includes imaginary persons ... In any case, madsrdf:PersonalName is clearly a subtype of rdfs:Label, and cannot map in any meaningful way to rda:Person.

bf:Person has the definition 'Individual or identity established by an individual (either alone or in collaboration with one or more other individuals)'. It is a subclass of foaf:Person, but there is no explanation of what the difference is. An 'individual' seems to be an RDA person because the entity has the ability to establish its own identity, which a legendary and fictitions 'person' does not; this is an indication of responsibility in the LRM sense. However, the BF definition seems to allow a pseudonym to be treated as a distinct 'person'. So I think it is best to assume that rda:Person is subclass of bf:Person and foaf:Person - and therefore we cannot safely use IRIs for instances of bf and foaf persons as subjects of an RDA predicate that has range rda:Person.

**Reply by GordonDunsire** [reply]
@CECSpecialistI: On place: Yes, an instance of a subclass is also an instance of the class. This is good for subjects such as 'Atlantic Ocean' and 'Ben Nevis' as well as 'United States' and 'Edinburgh'; that is, natural and man-made places.

**Reply by pennylenger** [reply]
> and therefore we cannot safely use IRIs for instances of bf and foaf persons as subjects of an RDA predicate that has range rda:Person.

@GordonDunsire I can't fully understand this sentence. Could you please give an example?

**Reply by pennylenger** [reply]
Thank you. Gordon. I found these about LCNAF and LCSH fictitious person 
Traditionally, all headings for fictitious characters and real, non-human entities1 were established in the Library of Congress Subject Headings (LCSH). However, Original RDA permitted names of individual fictitious characters to be used as descriptive access points when the character was credited with the creation of a work. In 2013, a revision to Original RDA made
it possible to establish all headings for individual fictitious characters in the NACO Name Authority File (NAF). Since then, all headings for individual fictitious characters and real, nonhuman entities have been established in the NAF and these name headings have been used as agents and subjects, as needed in bibliographic records.

In order to align RDA more closely with the Library Reference Model (LRM), the 3R Project changed the status of fictitious characters and real, non-human entities. The 2020 publication of the Official RDA Toolkit stated that responsibility for works could only be attributed to real human beings or collective agents. As a result of this change, the Program for Cooperative Cataloging (PCC) and the Library of Congress decided to adopt the following procedures for dealing with fictitious characters and real, non-human entities: (Throughout this guidance document, the phrase “fictitious characters and real, non-human entities” also includes the following types of personages: legendary or mythological beings, spirits, and deities.)
1. Name Authority Records (NARs) for fictitious characters and real, non-human entities
2. These NARs will receive a new description convention code (pccrda), indicating that the records conform to PCC’s implementation of RDA.
3. These NARs will also include an entity type designation.
4. As needed, fictitious characters and real, non-human entities may be credited with the creation of a work in bibliographic records.
5. These bibliographic records will receive a new description convention code (to be determined), indicating that the records conform to PCC’s implementation of RDA.
6. There is no change in the use of fictitious characters and real, non-human entities as subjects of works.

Establish fictitious characters and real, non-human entities in the NAF, following the same procedures for RDA Agents.
  - Do not use the description convention code “rda” on name authority records for fictitious characters and real, non-human entities. Instead, provide the new description convention code “pccrda”.
   - Provide an entity type designation.
      - PCC is developing an entity type vocabulary for use with these non-RDA Agents. This documentation will be updated when the vocabulary is implemented.

**Reply by GordonDunsire** [reply]
@pennylenger: This is interesting. I checked the NAR for 'Snoopy' and found one corresponding qualified name: 'Snoopy (Fictitious character)'. The MARC 21 authority record has the value 'rda' in field 040 subfield $e (Description conventions), which contradicts the LC/PCC guidance, perhaps because the source is 'UK'. Switching to id.loc.gov, the rwo for 'Snoopy (Fictitious character)' is typed as bf:Person and foaf:Person.

I also checked 'Sherlock Holmes", which has non-qualified name in LCNAF, etc." 'Holmes, Sherlock'. The rwo is also typed in id.loc.gov as bf:Person and foaf:Person. 'Fictitious characters' is given as a madsrdf:entityDescriptor, which is essentially a category of 'person'.

So I think, unfortunately, that the transform cannot use reliably the authority description convention code or name qualifiers to distinguish RDA persons from LC/BF/FOAF rwo persons.

**Reply by GordonDunsire** [reply]
@pennylenger: My previous comment 'and therefore we cannot safely use IRIs for instances of bf and foaf persons as subjects of an RDA predicate that has range rda:Person' means that an IRI for a rwo typed as bf:Person or foaf:Person, such as the 'Snoopy' and 'Sherlock Holmes' examples, cannot be used by the transform as a value of, say, rdamo:P30268 'has related person of manifestation' because the IRI may refer to a fictitious, legendary, or non-human entity.

**Comment by pennylenger** [comment]
Hi @AdamSchiff,  For LCSH, I initially thought that LCSH are all subjects which don't correspond to any RDA entities. However, it was mentioned in the meeting that within LCSH, the United States is defined as a geographic place. Is LCSH place equivalent to rda:Place?

And for LC Children's Subject Headings, LCDGT, LCGFT, LCMPT, MARC Genre Terms, MARC List of Languages, MARC Relators, is there any possibility that these sources contain entities similar to RDA entities? I can’t find a documentation explicitly defining or classifying their entity or class in a formal ontology like some other sources.

**Comment by pennylenger** [comment]
@GordonDunsire For RDA corporate body:A collective agent who is composed of persons who are organized for a common purpose or activity. We also treat meeting as corporate body in subject mapping. But for many other sources, their organization entity doesn't cover meeting. Do we treat organization from other sources as subclass of rda:corporate body?

**Reply by GordonDunsire** [reply]
@pennylenger: Yes, an external type that excludes meetings is a subclass of RDA Corporate Body. External values are therefore RDA-compatible :-)

**Comment by pennylenger** [comment]
I find some definitions about [NACO place and LCSH place](https://www.loc.gov/aba/rda/mgd/place/mg-place.pdf): Place is defined in RDA as “A given extent of space.” Not all places are established as name authority records in the LC/NACO Authority File. The Descriptive Cataloging Manual Z1, Appendix 1 and the Subject Headings Manual H 405 provide guidance on when certain kinds of places are established according to RDA/NACO in the LC/NAF and when they should be proposed through SACO as LC subject headings. SHM H 405 divides entities into two groups:
1) Entities always established according to descriptive cataloging conventions with authority records that always reside in the name authority file; 
2) Headings always established according to subject cataloging conventions with authority records that reside in the subject authority file if used only as subject headings; or reside in the name authority file if the headings are needed for use as descriptive access points. This MGD only applies to places in group 1 and to those in group 2 that are needed for use as descriptive access points (MARC 110 and 710 fields in bibliographic records).

**For jurisdictions, RDA instructions and LC-PCC policy statements for the place entity must sometimes be used in conjunction with those for the corporate body entity.**

In MARC authority records, authorized access points for places are tagged 151 and variant access points for places are tagged 451. When used as subjects in MARC bibliographic records, places are recorded in the 651 field. 

However, when places are used as access points in bibliographic records for creators of works or contributors to an expression, they are tagged as jurisdictions in 110 and 710 respectively. Authorized access points for subordinate jurisdictional corporate bodies (for example, government ministries, departments, legislative bodies, courts, and embassies of a country) are tagged 110 in MARC authority records, and as 110, 610, and 710 in MARC bibliographic records.

**Reply by GordonDunsire** [reply]
@pennylenger: So tag X10 indicator 1=1 'Jurisdiction name' maps to RDA Place, the same as with tag X51.

**Comment by AdamSchiff** [comment]
See also:

Q2221906  geographic location
Q98929991  place
Q618123   geographical feature

I think all of these are subclasses of Q27096213  geographic entity
which itself is a subclass of Q123349660  geolocatable entity

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by AdamSchiff** [comment]
The new PCC rules coding for use in 040 $e and entity coding for 075 has not been implemented yet.  It won't be until Official RDA is implemented. That's why fictitious persons and real non-humans are still coded as persons. We continue to follow original RDA and most PCC libraries will for several more years.  Implementation of official RDA by most PCC libraries won't happen until 2027 I think.

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
Box 352900
Seattle, WA 98195-2900
aschiff @ uw.edu
________________________________

**Comment by pennylenger** [comment]
[The PeriodO period ](https://perio.do/technical-overview/)defines historical periods that have an implicit or explicit association with a geographical region. Typically, a period has both a time and geographical information. For example: [Pre-Pottery Neolithic](https://client.perio.do/?page=period-view&backendID=web-https%3A%2F%2Fdata.perio.do%2F&authorityID=p08m57h&periodID=p08m57h9xrv)  10,200 BP – 8,000 BP | Levant and upper Mesopotamia. Is this equivalent to the RDA Timespan or RDA Place?

**Comment by pennylenger** [comment]
Similar question if for AAT, for example [Ancient Asian](https://www.getty.edu/vow/AATFullDisplay?find=beijing&logic=AND&note=&subjectid=300391434) (Asian, <styles, periods, and cultures by region>, Styles and Periods (hierarchy name)): Cultures and styles of any nations or peoples of Asia in ancient times; the time frame generally includes centuries BCE, sometimes including periods up to around 1000 CE.
It is under styles, periods, and cultures by region, but also has time and place information. Can it be mapped to RDA Timespan or RDA Place?

**Comment by AdamSchiff** [comment]
I think the periods in PeriodO represent timespans, even if they have a geographic aspect for some of them.  Not all periods would have a geographic aspect (e.g. Jurassic geological period).

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Reply by GordonDunsire** [reply]
I agree with @AdamSchiff.

I'm not sure that the PeriodO vocabulary is correctly encoded in RDF/SKOS. For example, there are multiple preferred labels for 'Neolithic' for different instances in the same SKOS scheme:

```
<http://n2t.net/ark:/99152/p0m64tdbthk>
skos:inScheme <http://n2t.net/ark:/99152/p0m64td>;
skos:prefLabel"Neolithic".
	
<http://n2t.net/ark:/99152/p0m64tdp3pt>
skos:inScheme <http://n2t.net/ark:/99152/p0m64td>;
skos:prefLabel "Neolithic".
```
That is, the preferred labels should be qualified by the geographic scope, but that's a problem for users of the vocabulary and not the transform.

**Comment by AdamSchiff** [comment]
I don't think these AAT are strictly timespan because they represent styles of art and style would not be covered by RDA at all.  I think they are subject concepts that include a chronological as well as geographical/cultural aspect.

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***

________________________________

**Reply by GordonDunsire** [reply]
I agree with @AdamSchiff. The example is typed as a http://vocab.getty.edu/ontology#Subject , http://www.w3.org/2004/02/skos/core#Concept, and http://vocab.getty.edu/ontology#Concept. We should treat AAT as a SKOS concept scheme.

**Reply by pennylenger** [reply]
Thank you Adam and Gordon. Should we check the different types of the concepts? Some hierarchy levels might correspond to RDA entities.
For example: for **AAT**
[people (agents)](https://www.getty.edu/vow/AATFullDisplay?find=Calligraphy&logic=AND&note=&subjectid=300024979)
Members of the species Homo sapiens and their close extinct relatives, as distinguished from other animals, spirits, or other entities.
Are terms under this hierarchy equivalent to the RDA Person?
[works (general, creative)](https://www.getty.edu/vow/AATFullDisplay?find=Calligraphy&logic=AND&note=&subjectid=300387357)
Objects, performances, or written or intellectual compilations, that are considered creative and require effort or labor.
Are terms under this hierarchy equivalent to the RDA work?
[organizations (groups)](https://www.getty.edu/vow/AATFullDisplay?find=Calligraphy&logic=AND&note=&page=1&subjectid=300025948)
Groups of people organized for a purpose, typically characterized by a more or less constant membership, a body of officers or functionaries, and a set of regulations governing their activities and conduct.
Are the terms under this hierarchy equivalent to the RDA organization?
[families (kinship groups)](https://www.getty.edu/vow/AATFullDisplay?find=Calligraphy&logic=AND&note=&subjectid=300055474)
Groups of individuals related by blood, marriage, adoption, or cohabitation.
Are the terms under this hierarchy equivalent to the RDA family?

For **TGN,** there are four types of concepts for place:
[**AdminPlaceConcept (OWL Class):**](http://vocab.getty.edu/ontology#AdminPlaceConcept )
Place defined by administrative boundaries and conditions, including inhabited places, nations, and empires. Used in TGN only.
Example: Burgundy region in France (TGN)
[**PhysPlaceConcept (OWL Class):**](http://vocab.getty.edu/ontology#PhysAdminPlaceConcept )
Physical feature, defined by its physical characteristics on planet Earth, including mountains, rivers, and oceans. Used in TGN only.
Example: Amazon River (TGN)
[**PhysAdminPlaceConcept (OWL Class):**](http://vocab.getty.edu/ontology#PhysAdminPlaceConcept )
Place that is both administrative and physical. Rarely used. Used in TGN only.
Example: 6003090 Kiik-Koba (TGN)
**Guide term:** Refers to records that serve as place savers to create a level in the hierarchy under which the TGN can collocate related places. They are enclosed in angled brackets.

For **ULAN**, can we treat [PersonConcept (OWL Class)](http://vocab.getty.edu/ontology#PersonConcept) as rda person?

**Reply by AdamSchiff** [reply]
If you open up the hierarchy under AAT, you can see that these concepts don't correspond to RDA entities.  The terms beneath aren't agents they are concepts.  For example for person:

![image](https://github.com/user-attachments/assets/79a3a602-8364-4c7c-96e3-dc14624b78df)

For organizations, the lower level terms are types of organizations, not corporate bodies:

![image](https://github.com/user-attachments/assets/b7e70fa4-a5f0-43b8-af76-bdcb96d0336c)

**Reply by pennylenger** [reply]
Thank you! Adam.

**Comment by pennylenger** [comment]
For below schemes which use SKOS concept, can we map some types of the concepts to rda entity? Or they are all concepts with no similar rad entity?
**Agriculture:**
AGROVOC: AGROVOC concepts are hierarchically organized under 25 general top concepts, such as: activities, processes, methods etc. 
[Humans](https://agrovoc.fao.org/browse/agrovoc/en/page/c_4586), [families](https://agrovoc.fao.org/browse/agrovoc/en/page/c_2785), [location](https://agrovoc.fao.org/browse/agrovoc/en/page/c_330988), [time](https://agrovoc.fao.org/browse/agrovoc/en/page/?uri=http%3A%2F%2Faims.fao.org%2Faos%2Fagrovoc%2Fc_7778)

**GACS Core:** Concept types. GACS Core distinguishes five types of concept: Chemical, Geographical, Organism, Product, and Topic
(GACS Core is not currently being updated and will not be actively maintained in the future.)

NAL Agricultural Thesaurus: NALT concept types (Organism, Chemical, Product, Geographical, Topic) express the most salient features of the concept space.
is Geographical concept equivalent to rda place?

**Map:** rda place?

**Pleiades:** four classes of information structures: places, locations, names, and connections.

**GeoNames:**
To define gn:Class as an ontology class intended for describing concept schemes, it’s explicitly defined as a subclass of SKOS’s ConceptScheme.
<owl:Class rdf:about="#Code">
    <rdfs:comment xml:lang="en">A feature code.</rdfs:comment>
    <rdfs:label xml:lang="en">Code</rdfs:label>
    <rdfs:subClassOf rdf:resource="http://www.w3.org/2004/02/skos/core#Concept"/>
  </owl:Class>
Three key ontology classes: Feature, Class and Code

**Reply by GordonDunsire** [reply]
@pennylenger: The entity skos:Concept is a supertype of most, if not all, things that are not predicates or properties. We can only break-down skos concepts into entity subtypes if:

- The IRI is based on a folder structure that isolates the subtype; for example ../places/...
- An instance description set says the IRI is the same as an IRI that is typed for the subtype.
- The subfolder documentation includes a definition that can be matched with an RDA entity definition; that is, the intension of the class indicates its semantics.
- The instances of concepts given as 'narrower', etc. of the subtype indicate the scope; that is, the extension of the class indicates its semantics.

These vocabularies of skos concepts generally lack this information, or the information shows that a mapping with an RDA entity is not safe.

For example, the narrower terms of 'Geographical' in the NALT vocabulary include generic subtypes of Place such as 'Mediterranean islands', which is not an RDA Place. I can't find any of the other potential indicators in NALT.

Pleiades place seems to be similar to RDA Place, but the conceptual overview states that 'Places are entirely abstract ... they have no spatial or temporal attributes of their own; that is, there is no class intension. It also says that place includes 'Spatial extents or thematic groupings of places defined by modern scholars ...'. A grouping of places is not, itself, a place in RDA unless it has a single spatial boundary. It is unclear if Pleiades conforms to this, so it is unsafe to map to RDA Place.

The Geonames Feature class seems to be close enough to RDA Place for the mapping to be safe.

**Comment by pennylenger** [comment]
@CECSpecialistI @lake44me @AdamSchiff @GordonDunsire @JianPLee I am trying to find the institution codes within the $0 for different sources. 
For example, 650 _7 |a Painting, Renaissance. |2 fast |0 (OCoLC)fst01051051
For fast, the code is "OCoLC"  
I have looked at some records, but I have not been able to find codes for other sources. We need these codes to create base URIs. Could you recommend any good methods for finding them?

**Comment by AdamSchiff** [comment]
The institution codes are at https://www.loc.gov/marc/organizations/
MARC Code List for Organizations (Network Development and MARC Standards Office, Library of Congress)<https://www.loc.gov/marc/organizations/>
MARC Code List for Organizations prepared by the Library of Congress Network Development and MARC Standards Office
www.loc.gov


Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Reply by pennylenger** [reply]
Many schemes are collaborative projects between different universities, organizations or library communities like [GAMECIP](https://www.wikidata.org/wiki/Property:P5371), [Homosaurus](https://homosaurus.org/v3) and [MusicBrainz](https://musicbrainz.org/). Does that mean we can't find an institution code for them?

**Comment by AdamSchiff** [comment]
I believe MARC also allows source codes in parentheses.

For example, I think this is valid:

380  Bisexual drama $2 homoit $0 https://homosaurus.org/v3/homoit0000149
as is this:
380  Bisexual drama $2 homoit $0 (homoit)homoit0000149

Adam

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by pennylenger** [comment]
In the [Répertoire International de Sources Musicales](https://rism.online/search?q=) (RISM) Online, the record format is quite similar to MARC. It has three types resource, person, and institution. A resource in RISM often represents a manuscript copy or autograph manuscript. Can we map the resource to RDA Item or Manifestation, but the record also contain Work or Expression information if it is an arrangement. Additionally, Can person and institution be mapped to RDA Person and Corporate Body?

**Reply by GordonDunsire** [reply]
@pennylenger: I see that the RISM ttl data references rism:Person, etc., but I can't find any definitions or an RISM ontology. There is some mention of using BIBFRAME. But the 'record' seems to be based on 'resource' (that is, a mix of WEMI), so I don't think it's safe to map. I also suspect that rism:Person, etc. are based on the broader semantics of foaf:Person, etc., and therefore it is also not safe to map the RDA entities. I must be missing something; I find it quite strange that the ontology behind RDF instance data is not formally defined.

**Comment by pennylenger** [comment]
The [Web NDL Authorities](https://id.ndl.go.jp/information/about_en/) use[ frbrent:Work](http://metadataregistry.org/schemaprop/show/id/794.html) which is deprecated. Is it equivalent to rda work?

**Reply by GordonDunsire** [reply]
@pennylenger: Yes. There is no essential difference between FRBR Work and LRM/RDA Work.

**Comment by pennylenger** [comment]
For Data Sources Without Linked Data in the bottom part of [Adam's table](https://www.wikidata.org/wiki/Wikidata:WikiProject_URIs_in_MARC), do we need to check their eneity type? And [Pleiades ID](https://www.wikidata.org/wiki/Property:P1584) also appear there.

**Reply by cspayne** [reply]
URIs are not available from those sources so I don't believe it's necessary.

**Comment by pennylenger** [comment]
@GordonDunsire I list below the sources that have entities equivalent to or subtypes of RDA entities after analysis. It may contain errors, please help review if you have time. Detailed analysis can be found in this [table](https://docs.google.com/spreadsheets/d/1p_lDvrR58zD7ZN0_QM1vaPvr06rvelA6cB0piCZB6es/edit?gid=0#gid=0):

1. CERL Thesaurus: FRBR:Person, FRBR:Corporate Body, FRBR:Place
2. BBC Things: cwork:CreativeWork, core:Organisation
3. DBpedia: dbo:Organization, dbo:Family
4. GeoNames: gn:Feature
5. FAST: schema:CreativeWork, schema:Organization
6. VIAF: schema:CreativeWork, schema:Organization
7. WorldCat Entities: wco:Organization, wco:Family, wco:Place 
8. Wikidata: human (Q5), organization (Q43229), family (Q8436), geographic entity (Q27096213), time interval (Q186081)
9. ORCID:person
10. GND (Gemeinsame Normdatei): Family
11. HMML Authority File: CreativeWork, Organization, Family, Expression
12. ISNI: schema:Organization
13. LC Children's Subject Headings: MADS/RDF NameTitle
14. LC/NACO Authority File: MADS/RDF NameTitle, BF Organization, BF Family, madsrdf:Geographic
15. LCSH: madsrdf:Geographic
16. MusicBrainz: work, Area
17. PeriodO: time
18. Répertoire International de Sources Musicales (RISM) Online: institution
19. Social Networks and Archival Context (SNAC): corpname, family
20. Web NDL Authorities: frbrent: Work, foaf:Organization, frbrent:Family

**Comment by pennylenger** [comment]
@hinxcode @cspayne @GordonDunsire This table [Approved URI](https://docs.google.com/spreadsheets/d/1ryMjPPLD6QyRx2vEruxqR7xWYNinaZjA1DWDuLr4ltI/edit?gid=0#gid=0) (it actually has not been reviewed and approved) put together entity type uri that can be used for lookup. There are some sources we have discussed and decided that it can be used but they don't have clear entity type uri such as  PeriodO. I put a note in the table. If anyone  can help check, that would be great!

**Comment by AdamSchiff** [comment]
Wikidata also has:

corporate body  http://www.wikidata.org/entity/Q106668099

Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Reply by pennylenger** [reply]
Thank you Adam. I have added this.

**Comment by GordonDunsire** [comment]
@pennylenger: I hope to find time to review the table over the next few days; you have gathered an impressive amount of data!

**Reply by pennylenger** [reply]
Thank you, Gordon. For the subclasses of approved entity types, such as Wikidata's 'geographic entity' (Q27096213), which are also RDA place, do we have an efficient method to include these subclasses in the lookup? It has too many subclasses.

**Reply by GordonDunsire** [reply]
@pennylenger: I have reviewed the table, after reformatting it for ease of use. I added some comments to the spreadsheet of LC sources as well. That spreadsheet was really helpful for making final approval!

I think that entities based on schema:CreativeWork are 'resource' entities that cannot be safely assigned to Work, Expression, or Manifestation, so I don't think they should be approved.

I assume you mean 'the lookup' of the table by the transform during processing. If the source's instance IRIs have a path structure that distinguishes subtypes, the folder/path root for each subtype can be a separate entry in the table. In the case of WikiData, there is no such structure in the instance IRIs, and we have to trust that the legacy data is valid for the tag and indicators for that specific RDA entity. So all WikiData instance IRIs may reference non-RDA entities, for example 'Person' instead of its subtype 'Human being', which is what maps to RDA Person.

I have left the approval of WikiData as a source of instance IRIs as a question-mark. If @AdamSchiff or others can assure us that PCC or other guidelines restrain the use of WD instances to 'approved' WD classes, we should regard these as unsafe and unapproved.

I don't think the transform can lookup the source in real time to determine the type of an instance IRI.

**Reply by hinxcode** [reply]
Dear team,

While Gordon, Adam, and Penny are still discussing whether WikiData can be approved as a source of instance IRIs, I'd like to share some technical details about the transformation process based on my recent face-to-face discussion with Penny. From a coder's perspective, I believe it is possible for the transform to look up the source in real time to determine the type of an instance IRI. By utilizing Java extensions within XSLT, we can implement the following workflow:

1. Get the domain of the resource's URI.
2. Fetch the resource's RDF model by domain.
3. Parse the RDF model and search for all "rdf:resource" attributes in the first-level property elements.
4. Retrieve the RDF details of the target by visiting the URIs obtained from the "rdf:resource"
5. (If needed) Find all superclasses recursively if the target has parents
6. Lookup "Approved URI" table by the target's classes and identify the target's RDA type accordingly.

I also made this figure, which should be much easier to read:
![Determining URI Flow Chart](https://github.com/user-attachments/assets/21077435-af0f-432c-b2d1-82641b1c8906)


This approach allows for real-time lookup and type determination, which could potentially address some of the concerns raised about WikiData and other sources. The draft version of the Java extension can be reviewed [here](https://github.com/uwlib-cams/MARC2RDA/blob/main/Working%20Documents/transformationCode/extension/src/main/java/edu/uwlib/cams/RdfPredicateExtractor.java), even though it is still under development.

I hope this technical perspective adds value to your ongoing discussions.

Best,
Ying-Hsiang

**Comment by pennylenger** [comment]
@AdamSchiff Hi Adam, could you please kindly review some rows that Gordon has left a question-mark? in the Approved column in this [table](https://docs.google.com/spreadsheets/d/1ryMjPPLD6QyRx2vEruxqR7xWYNinaZjA1DWDuLr4ltI/edit?gid=0#gid=0)? Thank you!

**Comment by AdamSchiff** [comment]
I added some comments to the table.

Adam


Adam L. Schiff
Principal Cataloger
University of Washington Libraries
(206) 543-8409
***@***.***
________________________________

**Comment by pennylenger** [comment]
@GordonDunsire @AdamSchiff @CECSpecialistI @cspayne I wanted to clarify whether Wikidata is approved. In today’s meeting (2024-11-13), Gordon mentioned that Wikidata is not approved, while Crystal said it is approved. Additionally, Adam has added some comments to the Wikidata rows in the approved URI table.

**Reply by cspayne** [reply]
I think the issue in phase 1 is that we cannot tell whether a Wikidata item is an approved type just by the IRI provided in the MARC data, so it is not safe to approve it.

**Reply by CECSpecialistI** [reply]
I understand now. We can't tell unless we query further whether the Wikidata item is an instance of Q5 human, and therefore it isn't safe for phase I. Bummer, man.

**Reply by GordonDunsire** [reply]
@CECSpecialistI and others: I think the methods we have decided to adopt will encourage communities and individuals to crowd-source the look-up in WD, because they will already be using WD for the output of the transform. With a stringified WD IRI, a look-up involves:

- copying the string
- pasting it into a browser
- examining the result for the type

If it turns out to be human, add statements to WD:

- Copy the 'has related person of work' statement, replace the datatype predicate with the object predicate, and replace the minted object IRI with the WD instance IRI
- Add an AAP statement to the WD IRI
- Add a sameAs statement for the minted and WD IRIs
- Optional: Copy the minted IRI AP statement and replace the minted IRI subject with the WD IRI - this can be easily entailed by a good linked open data application

I'm sure an enterprising grad student would be able to program a simple-to-use app for this :-)

And I think/hope that a lot of other things could be fixed in this way, so maybe not so much a bummer because it gets folks to 'own' the data.

**Reply by CECSpecialistI** [reply]
We can't substitute Wikidata IRI's for IRI's in our Wikibase Cloud instance because they are separate namespaces, but reconciliation with Wikidata is a great idea.

**Comment by cspayne** [comment]
Hi @GordonDunsire 
I had a follow-up question based on the slides you put together last week.
1. The slides discuss an 'approved' vs 'unapproved' source in $2. Why do we need to approve sources in $2, since these will indicate the source of the access point (and nomen), not the IRI?

**Reply by GordonDunsire** [reply]
@cspayne: The difference is that we use AAP for the minted entity from an approved source, and AP if the source is not approved.

If the source is not approved, we don't know what class of entity is being referenced. So we should not make the assumption that the heading is an AAP for the RDA entity, only the source entity. For example, assuming 'Holmes, Sherlock' is an AAP in its source and therefore is an AAP for who is effectively an unidentified person (a person who uses the pseudonym) is a 'harder' statement than stating it is an AP (which may or may not be applied to multiple distinct persons).

**Reply by cspayne** [reply]
Will the $2 source values be the same as the 024 or 3XX source codes in the [wikidata table](https://www.wikidata.org/wiki/Wikidata:WikiProject_URIs_in_MARC)? Or will more work need to be done to identify what the source codes for each source will be?

**Reply by GordonDunsire** [reply]
@cspayne: I assume the $2 values are the same as in the wd table.

**Reply by cspayne** [reply]
When the $2 is not approved, can we still mint a nomen and provide the scheme from $2? Just as an access point instead of an authorized one? Or should this just become a string value with no source provided?

**Reply by GordonDunsire** [reply]
@cspayne: I don't see why not; it is consistent with unapproved IRI sources.
